Backend-API/src/main/java/com/avangers/backendapi/controllers/PetController.java

```diff
@@ -23,10 +25,24 @@ public class PetController {
     private final UserService userService;
 
     @GetMapping("/my-pets")
-    public List<PetDTO> getMyPets(@AuthenticationPrincipal UserDetails userDetails) {
+
+    public List<PetDTO> getMyPets(Principal principal) {
         // gets UserDTO using the username (email)
-        FindUserResponseDTO user = userService.findUserByEmail(userDetails.getUsername());
+        FindUserResponseDTO user = userService.findUserByEmail(principal.getName());
         // Now use the user's ID to get the pets
         return petService.getPetsByOwnerId(Long.valueOf(user.getId()));
     }
```

Comment:
I think we should use the same version in every place - if we use ResponseEntity wrapper for ever other endpoint; maybe we should do it here as well.